src/libraries/System.Security.Cryptography/tests/MLDsaOpenSslTests.Unix.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Security.Cryptography.Tests
+{
+    [ConditionalClass(typeof(MLDsa), nameof(MLDsa.IsSupported))]
+    public sealed class MLDsaOpenSslTests : MLDsaTestsBase
+    {
+        protected override MLDsa GenerateKey(MLDsaAlgorithm algorithm)
+        {
+            using SafeEvpPKeyHandle key = Interop.Crypto.MLDsaGenerateKey(algorithm.Name, ReadOnlySpan<byte>.Empty);
+            return new MLDsaOpenSsl(key);
+        }
+
+        protected override MLDsa ImportPrivateSeed(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source)
+        {
+            SafeEvpPKeyHandle key = Interop.Crypto.MLDsaGenerateKey(algorithm.Name, source);
+            return new MLDsaOpenSsl(key);
+        }
+
+        protected override MLDsa ImportSecretKey(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source)
+        {
+            SafeEvpPKeyHandle key = Interop.Crypto.EvpPKeyFromData(algorithm.Name, source, privateKey: true);
+            return new MLDsaOpenSsl(key);
+        }
+
+        protected override MLDsa ImportPublicKey(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source)
+        {
+            SafeEvpPKeyHandle key = Interop.Crypto.EvpPKeyFromData(algorithm.Name, source, privateKey: false);
+            return new MLDsaOpenSsl(key);
+        }
+
+        [Fact]
+        public void MLDsaOpenSsl_Ctor_ArgValidation()
+        {
+            AssertExtensions.Throws<ArgumentNullException>("pkeyHandle", static () => new MLDsaOpenSsl(null));
+        }
+
+        [Fact]
+        public void MLDsaOpenSsl_WrongAlgorithm()
+        {
+            using RSAOpenSsl rsa = new RSAOpenSsl();
+            using SafeEvpPKeyHandle rsaHandle = rsa.DuplicateKeyHandle();
+            Assert.Throws<CryptographicException>(() => new MLDsaOpenSsl(rsaHandle));
+        }
+
+        [Fact]
+        public void MLDsaOpenSsl_DuplicateKeyHandle()
+        {
+            using SafeEvpPKeyHandle key = Interop.Crypto.EvpKemGeneratePkey(MLDsaAlgorithm.MLDsa44.Name);
```

Comment:
This like looks like a copy-paste error. It seems like it should be `MLDsaGenerateKey` if you wanted ML-DSA or it should be the right ML-KEM algo name if you really wanted ML-KEM.

I understand why this test is not failing (all the crypto algos go through `EVP_PKEY_CTX_new_from_name`/`EVP_PKEY_keygen` for key generation), but having said that, I really would have expected it to fail since it's mixing DSA and KEM. I suppose no one else is going to use `Interop.Crypto.EvpKemGeneratePkey` directly so we don't need to guard against this.